frontend/src/api/k8s/__tests__/pvcs.spec.ts

```diff
@@ -0,0 +1,300 @@
+import {
+  K8sStatus,
+  k8sCreateResource,
+  k8sDeleteResource,
+  k8sGetResource,
+  k8sListResourceItems,
+  k8sPatchResource,
+  k8sUpdateResource,
+} from '@openshift/dynamic-plugin-sdk-utils';
+import { mock200Status, mock404Error } from '~/__mocks__/mockK8sStatus';
+import { mockPVCK8sResource } from '~/__mocks__/mockPVCK8sResource';
+import {
+  assemblePvc,
+  createPvc,
+  deletePvc,
+  getAvailableMultiUsePvcs,
+  getDashboardPvcs,
+  getPvc,
+  updatePvc,
+  updatePvcSize,
+} from '~/api/k8s/pvcs';
+import { PersistentVolumeClaimKind } from '~/k8sTypes';
+import { CreatingStorageObject } from '~/pages/projects/types';
+
+jest.mock('@openshift/dynamic-plugin-sdk-utils', () => ({
+  k8sGetResource: jest.fn(),
+  k8sListResourceItems: jest.fn(),
+  k8sCreateResource: jest.fn(),
+  k8sUpdateResource: jest.fn(),
+  k8sDeleteResource: jest.fn(),
+  k8sPatchResource: jest.fn(),
+}));
+
+const k8sGetResourceMock = jest.mocked(k8sGetResource<PersistentVolumeClaimKind>);
+const k8sListResourceItemsMock = jest.mocked(k8sListResourceItems<PersistentVolumeClaimKind>);
+const k8sCreateResourceMock = jest.mocked(k8sCreateResource<PersistentVolumeClaimKind>);
+const k8sUpdateResourceMock = jest.mocked(k8sUpdateResource<PersistentVolumeClaimKind>);
+const k8sDeleteResourceMock = jest.mocked(k8sDeleteResource<PersistentVolumeClaimKind, K8sStatus>);
+const k8sPatchResourceMock = jest.mocked(k8sPatchResource<PersistentVolumeClaimKind>);
+
+const data: CreatingStorageObject = {
+  nameDesc: {
+    name: 'pvc',
+    description: 'Test Storage',
+  },
+  size: '5Gi',
+};
+
+const assemblePvcResult: PersistentVolumeClaimKind = {
+  apiVersion: 'v1',
+  kind: 'PersistentVolumeClaim',
+  metadata: {
+    annotations: {
+      'openshift.io/description': 'Test Storage',
+      'openshift.io/display-name': 'pvc',
+    },
+    labels: { 'opendatahub.io/dashboard': 'true' },
+    name: 'pvc',
+    namespace: 'namespace',
+  },
+  spec: {
+    accessModes: ['ReadWriteOnce'],
+    resources: { requests: { storage: '5Gi' } },
+    volumeMode: 'Filesystem',
+  },
+  status: { phase: 'Pending' },
+};
+
+const createAssemblePvcs = (accessModes: string[]) => ({
+  ...assemblePvcResult,
+  spec: { ...assemblePvcResult.spec, accessModes },
+});
+describe('assemblePvc', () => {
+  it('should assemble pvc without editName', () => {
+    const result = assemblePvc(data, 'namespace');
+    expect(result).toStrictEqual(assemblePvcResult);
+  });
+
+  it('should assemble pvc with editName', () => {
+    const result = assemblePvc(data, 'namespace', 'editName');
+    expect(result).toStrictEqual({
+      ...assemblePvcResult,
+      metadata: { ...assemblePvcResult.metadata, name: 'editName' },
+    });
+  });
+});
+
+describe('getPvc', () => {
+  it('should fetch and return pvc', async () => {
+    k8sGetResourceMock.mockResolvedValue(mockPVCK8sResource({ uid: 'test' }));
```

Comment:
Can you create mock once and use  it in other part ?, this will prevent adding extra uid argument and changing the mock function